engine/src/main/java/com/odysseusinc/arachne/executionengine/execution/r/DockerService.java

```diff
@@ -0,0 +1,268 @@
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+@Slf4j
+public class DockerService extends RService {
+    @Autowired
+    @Qualifier("analysisTaskExecutor")
+    private ThreadPoolTaskExecutor executor;
+    @Autowired
+    private DockerProperties dockerProperties;
+
+    private static final String DOCKER_ENV_APP_DEBUG = "RUNTIME_ENV_APP_DEBUG";
+    private static final String DOCKER_ENV_MEMORY_LIMIT_KEY = "RUNTIME_ENV_DOCKER_MEMORY_LIMIT";
+    private static final String DOCKER_ENV_CPU_LIMIT_KEY = "RUNTIME_ENV_DOCKER_CPU_LIMIT";
+    private static final String DOCKER_ENV_MEMORY_LIMIT_VALUE = "512m";
+    private static final String DOCKER_ENV_CPU_LIMIT_VALUE = "0.5";
+
+    @Override
+    public Overseer analyze(
+            AnalysisSyncRequestDTO analysis, File file, DescriptorBundle bundle,
+            KrbConfig krbConfig, BiConsumer<String, String> callback, Integer updateInterval) {
+        Instant started = Instant.now();
+        StringBuffer stdout = new StringBuffer();
+        BiConsumer<String, String> callbackWithStdOut = callback.andThen((stage, out) -> {
+            stdout.append(out);
+        });
+
+        CompletableFuture<ExecutionOutcome> future = CompletableFuture.supplyAsync(() -> {
+            try (DockerClient client = dockerClient()) {
+                String image = analysis.getDockerImage();
+                pullImageIfNotExists(image);
+                log.info("Creating container for image: {} and analysis Id: {}", image, analysis.getId());
+                CreateContainerResponse container = client.createContainerCmd(image)
+                        .withStdinOpen(true)
+                        .withTty(true)
+                        .exec();
+                log.info("Container created for image {} with container id {}", image, container.getId());
+                String containerId = container.getId();
+                runContainer(client, containerId, file, analysis, callbackWithStdOut);
+                remove(client, containerId);
+                return new ExecutionOutcome(Stage.COMPLETED, null, "");
+            } catch (IOException e) {
+                log.error("Execution [{}] failed:", analysis.getId(), e);
+                callbackWithStdOut.accept(Stage.EXECUTE, e.getMessage());
+                return new ExecutionOutcome(Stage.COMPLETED, e.getMessage(), "");
+            }
+        }, executor);
+        return new SQLService.SqlOverseer(analysis.getId(), started, stdout, future);
+    }
+
+    private void runContainer(DockerClient dockerClient, String containerId, File analysisDir, AnalysisSyncRequestDTO analysis, BiConsumer<String, String> callback) {
+        String execFileName = analysis.getExecutableFileName();
+
+        log.info("Starting container: {}", containerId);
+        dockerClient.startContainerCmd(containerId).exec();
+
+        String codeDir = "/bin";
```

Comment:
Why `/bin`? Shouldn't we have dedicated directory for analysis? Or at least `/etc/smth`

Reply:
`/bin` is always present in all images while other folder would need to be created or an error could occur.